common/static/common/js/utils/page_factory.js

```diff
@@ -0,0 +1,13 @@
+define([], function () {
+    return function invokePageFactory(factory) {
+        if (typeof window.pageFactoryArguments == "undefined") {
+            throw "window.pageFactoryArguments must be initialized before calling invokePageFactory. Use the <%static:require_page> template tag.";
+        }
+        args = window.pageFactoryArguments[factory.name];
+
+        if (typeof args == "undefined") {
+            throw "window.pageFactoryArguments[\"" + factory.name + "\"] must be initialized before calling invokePageFactory. Use the <%static:require_page> template tag."
```

Comment:
You can use an es6 [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here -- a little cleaner than concatenation.